### PR TITLE
improvement project added in each level

### PR DESCRIPTION
**Pull request by aman-tunerlabs**

### CODE REVIEW

File: controllers/v1/criteriaController.js [hunk 1]
```diff
@@ -65,127 +67,17 @@ module.exports = class Criteria extends Abstract {
           });
         }());
 
+        let updatedCriteria = await criteriasHelper.upload(
```
Comment (aks30) [inline controllers/v1/criteriaController.js:70]: @aman-tunerlabs - Can we rename it to criteriaHelper just to maintain consistency in naming helpers with module name. You might have to replace all across

File: controllers/v1/observationsController.js [hunk 2]
```diff
@@ -539,7 +538,16 @@ module.exports = class Observations extends Abstract {
...
 
-                let entityDocuments = await entitiesHelper.search(observationDocument.entityTypeId, req.searchText, req.pageSize, req.pageNo);
+                let entityDocuments = 
+                await entitiesHelper.search(
+                    observationDocument.entityTypeId, 
+                    req.searchText, 
+                    req.pageSize, 
+                    req.pageNo,
+                    false,
+                    req.query.schoolTypes,
```
Comment (aks30) [inline controllers/v1/observationsController.js:548]: @aman-tunerlabs - We should not have schoolTypes and administrationTypes coming from query or request. It is taken from the logged in user profile

File: module/criteria/helper.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+let unnatiService = require(ROOT_PATH+"/generics/services/unnati")
```
Comment (aks30) [inline module/criteria/helper.js:1]: @aman-tunerlabs - Lets rename it to improvement project service, unnati is just a name and diff states can have different app names

File: module/criteria/helper.js [hunk 4]
```diff
@@ -132,4 +134,233 @@ module.exports = class criteriaHelper {
...
+              parsedCriteria[parsedCriteriaKeys[key]].split(",");
+
+              improvementProjectIds = _.concat(improvementProjectIds,improvements);
+
+            }
+          }
+
+        }
+
+        let improvementProjects = await unnatiService.getImprovementProjects(
```
Comment (aks30) [inline module/criteria/helper.js:182]: @aman-tunerlabs - Perform API call only if improvementProjectIds is not empty

File: module/submissions/helper.js [hunk 5]
```diff
@@ -550,6 +550,877 @@ module.exports = class SubmissionsHelper {
         return answer;
     }
 
+    /**
```
Comment (aks30) [inline module/submissions/helper.js:553]: @aman-tunerlabs - This has moved to the scoring module, request you to make changes there.

File: module/entities/helper.js [hunk 6]
```diff
@@ -795,7 +795,15 @@ module.exports = class EntitiesHelper {
...
 
-    static search(entityTypeId, searchText, pageSize, pageNo, entityIds = false) {
+    static search(
+        entityTypeId, 
+        searchText,
+        pageSize, 
+        pageNo, 
+        entityIds = false,
+        aclData = [],
+        administrationTypes = ""
```
Comment (aks30) [inline module/entities/helper.js:805]: @aman-tunerlabs - Is administrationTypes required still ?

File: module/entities/helper.js [hunk 7]
```diff
@@ -808,6 +816,11 @@ module.exports = class EntitiesHelper {
                     queryObject["$match"]["_id"]["$in"] = entityIds;
                 }
 
+                if( aclData.length > 0 ) {
```
Comment (aks30) [inline module/entities/helper.js:819]: @aman-tunerlabs - Has the same filter added in user extension get profile APIs ?

Comment (aman-tunerlabs) [reply module/entities/helper.js:819]: @aks30  for get profile api we just added one extra field called as acl in response.

**Pull request merged**